pb/metainfo.proto

```diff
@@ -781,10 +786,45 @@ message ObjectFinishMoveRequest {
     repeated EncryptedKeyAndNonce new_segment_keys = 5;
 }
 
+message ObjectFinishMoveResponse { }
+
+// Server side copy.
+// Has the same fields as move.
+message ObjectBeginCopyRequest {
+    RequestHeader header = 15;
+
+    bytes source_bucket = 1;
+    bytes source_encrypted_object_key = 2;
+
+    bytes destination_bucket = 3;
+    bytes destination_encrypted_object_key = 4;
+}
+
+message ObjectBeginCopyResponse {
+    bytes destination_stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
```

Comment:
why "destination_stream_id"? should this be source_stream_id

Reply:
Ok I misunderstood the Move protocol. Changed. Explicit naming is good :).

Do you know why destination bucket and key are in Begin and also Finish?

Reply:
> Do you know why destination bucket and key are in Begin and also Finish?

as far I remember for BeginMoveObject its mainly for validation purposes e.g. check access permissions for destination location or destination bucket name validation.